src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1167 @@
+  else if (app_name == APP_NAME_CSQL || app_name == APP_NAME_LOADDB)
+    {
+      retval =
+	snprintf (filename_buf, buf_size, "%s/%s/%s_%s_ddl.log", env_root, DDL_LOG_PATH,
+		  cub_ddl_log_get_app_name (app_name), ddl_audit_handle->db_name);
+    }
+  else
+    {
+      filename_buf[0] = '\0';
+      return -1;
+    }
+
+  if (retval < 0)
+    {
+      assert (false);
+      filename_buf[0] = '\0';
+    }
+  return retval;
+}
+
+static FILE *
+cub_ddl_log_open (T_APP_NAME app_name)
+{
+  FILE *fp;
+  char *tpath = NULL;
+  int len;
+
+  len = cub_ddl_log_make_filename (ddl_audit_handle->log_filepath, PATH_MAX, app_name);
+
+  if (ddl_audit_handle->log_filepath[0] == '\0' || len < 0)
+    {
+      goto file_error;
+    }
+
+  if (cub_ddl_log_create_dir (ddl_audit_handle->log_filepath) < 0)
+    {
+      goto file_error;
+    }
+
+  /* note: in "a+" mode, output is always appended */
+  fp = cub_ddl_log_fopen_and_lock (ddl_audit_handle->log_filepath, "a+");
+
+  return fp;
+
+file_error:
+  return NULL;
+}
+
+void
+cub_ddl_log_write_end ()
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  if (prm_get_bool_value (PRM_ID_DDL_AUDIT_LOG) == FALSE)
+    {
+      goto ddl_log_free;
+    }
+
+  if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+    {
+      if (ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_SCHEMA
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_INDEX
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_TRIGGER)
+	{
+	  goto ddl_log_free;
+	}
+    }
+  else
+    {
+      if (cub_ddl_log_is_ddl_type (ddl_audit_handle->stmt_type) == FALSE)
+	{
+	  goto ddl_log_free;
+	}
+    }
+  if (ddl_audit_handle->execute_type == DDL_LOG_RUN_EXECUTE_FUNC)
+    {
+      cub_ddl_log_write ();
+    }
+
+ddl_log_free:
+  cub_ddl_log_free (false);
+}
+
+void
+cub_ddl_log_write ()
+{
+  FILE *fp;
+  char buf[DDL_LOG_BUFFER_SIZE] = { 0 };
+  char dest_path[PATH_MAX] = { 0 };
+  int len = 0;
+  int ret = 0;
+
+  if (ddl_audit_handle == NULL || prm_get_bool_value (PRM_ID_DDL_AUDIT_LOG) == FALSE)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+    {
+      if (ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_SCHEMA
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_INDEX
+	  && ddl_audit_handle->loaddb_file_type != LOADDB_FILE_TYPE_TRIGGER)
+	{
+	  return;
+	}
+    }
+  else
+    {
+      if (cub_ddl_log_is_ddl_type (ddl_audit_handle->stmt_type) == FALSE)
+	{
+	  return;
+	}
+    }
+
+  fp = cub_ddl_log_open (ddl_audit_handle->app_name);
+
+  if (fp != NULL)
+    {
+      if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+	{
+	  if (cub_ddl_log_make_schema_file_name (ddl_audit_handle->file_name, dest_path, PATH_MAX) < 0)
+	    {
+	      goto write_error;
+	    }
+	}
+
+      len = cub_ddl_log_create_lgo_mgs (buf);
+      if (len < 0)
+	{
+	  goto write_error;
+	}
+
+      if (fwrite (buf, sizeof (char), len, fp) != len)
+	{
+	  goto write_error;
+	}
```

Comment:
Merge two **if**s

```
if (len < 0 || fwrite (buf, sizeof (char), len, fp) != len)
  {
    goto write_error;
  }
```

Reply:
Thank you, i will change code as your recommend.